python/lsst/ts/mtdome/power_management/power_management_handler.py

```diff
@@ -0,0 +1,486 @@
+        priorities, from high to low, are:
+
+          * Open/close the shutter.
+          * Move/crawl the light/wind screen.
+          * Open/close the louvers.
+          * Enable the fans.
+
+        If any higher priority command in scheduled and any lower command is
+        currently being executed, the motion resulting from the lower priority
+        command first is stopped. Only then the higher priority command is
+        issued.
+        If any lower priority command is scheduled then it will need to wait
+        for the higher priority ones to finish.
+        Any other command will be scheduled.
+
+        Parameters
+        ----------
+        current_power_draw : `dict`[`LlcName`, `float`]
+            Dict of the current power draw [W] for the subsystems, or Lower
+            Level Components.
+
+        Returns
+        -------
+        ScheduledCommand | None
+            The next command to be issued, or None if no commands currently can
+            be issued or are scheduled.
+
+        Notes
+        -----
+        It is assumed that the Rear Access Door, the Overhead Bridge Crane and
+        the Calibration screen are not used in this mode.
+        """
+        _, scheduled_command = await self.command_queue.get()
+        match scheduled_command.command:
+            case CommandName.OPEN_SHUTTER | CommandName.CLOSE_SHUTTER | CommandName.SEARCH_ZERO_SHUTTER:
+                return await self.issue_scheduled_command_or_stop_commands(
+                    scheduled_command,
+                    current_power_draw,
+                    [STOP_EL, STOP_LOUVERS, STOP_FANS],
+                    [],
+                )
+            case CommandName.MOVE_EL | CommandName.CRAWL_EL:
+                return await self.issue_scheduled_command_or_stop_commands(
+                    scheduled_command,
+                    current_power_draw,
+                    [STOP_LOUVERS, STOP_FANS],
+                    [LlcName.APSCS],
+                )
+            case CommandName.SET_LOUVERS | CommandName.CLOSE_LOUVERS:
+                return await self.issue_scheduled_command_or_stop_commands(
+                    scheduled_command,
+                    current_power_draw,
+                    [STOP_FANS],
+                    [LlcName.APSCS, LlcName.LWSCS],
+                )
+            case CommandName.FANS:
+                # Switching off the fans is always allowed.
+                if scheduled_command.params["action"] == OnOff.OFF:
+                    return scheduled_command
+                else:
+                    return await self.issue_scheduled_command_or_stop_commands(
+                        scheduled_command,
+                        current_power_draw,
+                        [],
+                        [LlcName.APSCS, LlcName.LWSCS, LlcName.LCS],
+                    )
+            case _:
+                return scheduled_command
+
+    async def get_next_command_maintenance(
```

Comment:
See my comment above about using "get_<something>" on the name of this method.

Reply:
See my reply above.